src/main/java/Parser.java

```diff
@@ -0,0 +1,130 @@
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Parser {
+    protected Storage file;
+    protected TaskList tList;
+
+    /**
+     * Constructor to init a Parser class.
+     * @param file
+     * @param tList
+     */
+    public Parser(Storage file, TaskList tList) {
+        this.file = file;
+        this.tList = tList;
+    }
+
+    /**
+     * Method to process the different inputs that are taken in.
+     * @throws DukeException
+     * @throws IOException
+     */
+    public void parse() throws DukeException, IOException {
+        ArrayList<Task> arr = tList.getArr();
+        // Main logic, take in input from scanner and process accordingly
+        Scanner sc = new Scanner(System.in);
+        while (true) {
```

Comment:
I think you can use a switch statement (and perhaps Enums as well) to make your code more elegant.